src/components/lookupPublicKey/useLookupPublicKey.ts

```diff
@@ -0,0 +1,117 @@
+import { ethers, UnsignedTransaction } from 'ethers';
+import { TransactionResponse } from '@ethersproject/providers';
+import { useState } from 'react';
+import axios from 'axios';
+import { useProvider, useNetwork } from 'wagmi';
+
+export enum LookupPublicKeyStatus {
+  SUCCESS,
+  ERROR,
+  NO_TRANSACTIONS,
+  LOADING,
+  INVALID_ADDRESS,
+  WRONG_NETWORK,
+  NOT_CONNECTED,
+}
+
+const etherscanEndpoint = 'https://api.etherscan.io/api';
+const etherscanApikey = process.env.REACT_APP_ETHERSCAN_APIKEY;
+const getParameters =
+  'module=account&action=txlist&startblock=0&endblock=99999999&page=1&offset=1000&sort=asc';
+
+export function useLookupPublicKey() {
+  const [lookupStatus, setLookupStatus] = useState<LookupPublicKeyStatus | null>(null);
+  const provider = useProvider();
+  const { chain } = useNetwork();
+
+  async function recoverPublicKeyFromTransaction(transactionIn: TransactionResponse) {
+    const transactionOut: UnsignedTransaction = {
+      type: transactionIn.type,
+      nonce: transactionIn.nonce,
+      gasLimit: transactionIn.gasLimit,
+      to: transactionIn.to,
+      value: transactionIn.value,
+      data: transactionIn.data,
+
+      ...(transactionIn.chainId && { chainId: transactionIn.chainId }),
+
+      ...((transactionIn.type === 0 || transactionIn.type === 1 || !transactionIn.type) && {
```

Comment:
If there is some way to label these conditionals that makes sense (which there may not be), would recommend breaking them out into their own named functions for easier readability, i.e.

```
function isPre1559Tx(type): boolean {
  return (!type || transactionIn.type === 1)
}

...(isPre1559Tx(transactionIn.type) && {
   gasPrice: transactionIn.gasPrice,
 }),
```

Reply:
Thought of that too ... and then forgot to do it ... thanks